ballerina/tests/from_xml_test.bal

```diff
@@ -0,0 +1,628 @@
+                                <country>Sri Lanka</country>
+                            </address>
+                            <codes>
+                                <item>4</item>
+                                <item>8</item>
+                                <item>9</item>
+                            </codes>
+                        </BookStore5>
+                        <!-- some comment -->
+                        <?doc document="book.doc"?>`;
+    BookStore5 expected = {
+        storeName: xml `foo`,
+        postalCode: 94,
+        isOpen: true,
+        address: xml `<street>Galle Road</street><city>Colombo</city><country>Sri Lanka</country>`,
+        codes: xml `<item>4</item><item>8</item><item>9</item>`,
+        'xmlns\:ns0: "http://sample.com/test",
+        status: "online"
+    };
+    BookStore5 actual = check fromXml(payload);
+    test:assertEquals(actual, expected, msg = "testRecordToXml5 result incorrect");
+}
+
+xml xmValue = xml `<Bill xmlns="example.com" attr="attr-val" xmlns:ns="ns.com" ns:attr="ns-attr-val">
+                <PurchesedItems>
+                    <PLine><ItemCode>223345</ItemCode><Count>10</Count></PLine>
+                    <PLine><ItemCode>223300</ItemCode><Count>7</Count></PLine>
+                    <PLine><ItemCode discount="22%">200777</ItemCode><Count>7</Count></PLine>
+                </PurchesedItems>
+                <Address xmlns="">
+                    <StreetAddress>20, Palm grove, Colombo 3</StreetAddress>
+                    <City>Colombo</City>
+                    <Zip>300</Zip>
+                    <Country>LK</Country>
+                </Address>
+              </Bill>`;
+
+type Bill record {
+    PurchesedItems1 PurchesedItems;
+    Address10 Address;
+    string 'xmlns?;
+    string 'xmlns\:ns?;
+    string attr?;
+    string ns\:attr?;
+};
+
+type PurchesedItems1 record {
+    Purchase1[] PLine;
+};
+
+type Purchase1 record {
+    string|ItemCode1 ItemCode;
+    int Count;
+};
+
+type ItemCode1 record {
+    string discount;
+    string \#content;
+};
+
+type Address10 record {
+    string StreetAddress;
+    string City;
+    int Zip;
+    string Country;
+    string 'xmlns?;
+};
+
+@test:Config {
+    groups: ["fromXml"]
+}
+function testComplexXmlElementToRecord() returns error? {
+    Bill expected = {
+        PurchesedItems: {
+            PLine: [
+                {ItemCode: "223345", Count: 10},
+                {ItemCode: "223300", Count: 7},
+                {
+                    ItemCode: {discount: "22%", \#content: "200777"},
+                    Count: 7
+                }
+            ]
+        },
+        Address: {
+            StreetAddress: "20, Palm grove, Colombo 3",
+            City: "Colombo",
+            Zip: 300,
+            Country: "LK",
+            'xmlns: ""
+        },
+        'xmlns: "example.com",
+        'xmlns\:ns: "ns.com",
+        attr: "attr-val",
+        ns\:attr: "ns-attr-val"
```

Comment:
This is not correct. We cannot map values of attributes if user doesn't mark fields as attributes using annotations.

Reply:
Fixed in [821c3fb](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/431/commits/821c3fb1aaf2e91242ee44fc8e5fcdceabc54099)